pkg/antctl/raw/set/flowaggregator/command_test.go

```diff
@@ -0,0 +1,294 @@
+		configMapName  string
+		podNamespace   string
+		args           []string
+		expectedErr    string
+		expectedOutput map[string]string
+	}{
+		{
+			name:           "valid ConfigMap and nil args",
+			configMapName:  "testConfigMap",
+			podNamespace:   "flow-aggregator",
+			expectedOutput: map[string]string{"flow-aggregator.conf": "{}\n"},
+		},
+		{
+			name:          "invalid ConfigMap and nil args",
+			configMapName: "",
+			podNamespace:  "flow-aggregator",
+			expectedErr:   "failed to locate flow-aggregator-config ConfigMap volume",
+		},
+		{
+			name:           "valid ConfigMap and valid args",
+			configMapName:  "testConfigMap",
+			podNamespace:   "flow-aggregator",
+			args:           []string{"clickHouse.enable=true"},
+			expectedOutput: map[string]string{"flow-aggregator.conf": "clickHouse:\n  enable: true\n"},
+		},
+		{
+			name:          "valid ConfigMap and invalid args",
+			configMapName: "testConfigMap",
+			podNamespace:  "flow-aggregator",
+			args:          []string{"clickhouse=true"},
+			expectedErr:   "unknown configuration parameter, please check antctl set flow-aggregator -h",
+		},
+		{
+			name:          "valid ConfigMap and invalid args syntax",
+			configMapName: "testConfigMap",
+			podNamespace:  "flow-aggregator",
+			args:          []string{"clickhouse.enable=true clickhopuse.debug=true"},
+			expectedErr:   "query should contain exactly one '='",
+		},
+		{
+			name:          "wrong Pod Namespace",
+			configMapName: "testConfigMap",
+			podNamespace:  "flow-visibility",
+			expectedErr:   "request namespace does not match object namespace",
+		},
+	}
+
+	getClients = func(cmd *cobra.Command) (kubernetes.Interface, error) {
+		return k8sClient, nil
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			os.Setenv("POD_NAMESPACE", tc.podNamespace)
+			os.Setenv("FA_CONFIG_MAP_NAME", tc.configMapName)
+			defer os.Unsetenv("POD_NAMESPACE")
+			defer os.Unsetenv("FA_CONFIG_MAP_NAME")
+			err := updateRunE(cmd, tc.args)
+			if err != nil {
+				assert.Contains(t, err.Error(), tc.expectedErr)
+			} else {
+				cm, _ := k8sClient.CoreV1().ConfigMaps(os.Getenv("POD_NAMESPACE")).Get(context.TODO(), fakeConfigMap.Name, metav1.GetOptions{})
+				assert.Equal(t, cm.Data, tc.expectedOutput)
+			}
+		})
+	}
+}
+
+func TestSetBoolOrFail(t *testing.T) {
+	tcs := []struct {
+		name          string
+		value         string
+		expectedValue bool
+	}{
+		{
+			name:          "true boolean value",
+			value:         "1",
+			expectedValue: true,
+		},
+		{
+			name:          "false boolean value",
+			value:         "f",
+			expectedValue: false,
+		},
+		{
+			name:  "invalid value",
+			value: "tt",
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			var actualValue bool
+			err := setBoolOrFail(&actualValue, tc.value)
+			if err != nil {
+				assert.Contains(t, err.Error(), "invalid syntax")
+			} else {
+				assert.Equal(t, tc.expectedValue, actualValue)
+			}
```

Comment:
ditto, it should determine what to check based on expected result, instead of actual result, otherwise the test would succeed anyway if `setBoolOrFail` doesn't function correctly. For example:
```
func setBoolOrFail(b *bool, value string) error {
	return fmt.Errorf("invalid syntax")
}
```
would also pass your tests.